core/src/test/java/org/apache/calcite/test/TopDownOptTest.java

```diff
@@ -65,92 +70,177 @@
  * </ol>
  */
 class TopDownOptTest extends RelOptTestBase {
-
-  protected DiffRepository getDiffRepos() {
-    return DiffRepository.lookup(TopDownOptTest.class);
-  }
-
-  Sql sql(String sql) {
-    VolcanoPlanner planner = new VolcanoPlanner();
-    // Always use top-down optimization
-    planner.setTopDownOpt(true);
-    planner.addRelTraitDef(ConventionTraitDef.INSTANCE);
-    planner.addRelTraitDef(RelCollationTraitDef.INSTANCE);
-
-    RelOptUtil.registerDefaultRules(planner, false, false);
-
-    // Keep deterministic join order
-    planner.removeRule(JoinCommuteRule.INSTANCE);
-    planner.removeRule(JoinPushThroughJoinRule.LEFT);
-    planner.removeRule(JoinPushThroughJoinRule.RIGHT);
-
-    // Always use merge join
-    planner.removeRule(EnumerableRules.ENUMERABLE_JOIN_RULE);
-    planner.removeRule(EnumerableRules.ENUMERABLE_CALC_RULE);
-
-    // Always use sorted agg
-    planner.removeRule(EnumerableRules.ENUMERABLE_AGGREGATE_RULE);
-    planner.addRule(EnumerableRules.ENUMERABLE_SORTED_AGGREGATE_RULE);
```

Comment:
These 2 might be better in the default rule setting of `Query`, always use sorted agg in our tests.